rules-tests/TypeDeclaration/Rector/ClassMethod/StrictArrayParamDimFetchRector/Fixture/cover_isset_with_dim_fetch.php.inc

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\StrictArrayParamDimFetchRector\Fixture;
+
+final class CoverIssetWithDimFetch
+{
+    public function run($param, $item): bool
+    {
+        if (! isset($param[$item])) {
+            return false;
+        }
+
+        if ($param[$item] === 100) {
+            return true;
+        }
+
+        return false;
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\ClassMethod\StrictArrayParamDimFetchRector\Fixture;
+
+final class CoverIssetWithDimFetch
+{
+    public function run(array $param, $item): bool
```

Comment:
in this case $param could be anything which implements ArrayAccess not just the rare case of string-offset access

Reply:
Indeed, it's a choice of practice over risk of few edge cases. We apply same logic to other rules too, to ease manual daunting work on project upgrades.

If your codebase uses any type in this case, exclude the rule.